packages/webapp/src/_app/eos/ual/softkey/ual-softkey-login-modal.tsx

```diff
@@ -0,0 +1,84 @@
+import React from "react";
+import { PrivateKey } from "eosjs/dist/eosjs-jssig";
+
+import { generateEncryptionKey } from "_app/eos";
+import { useFormFields } from "_app/hooks";
+import { Button, Form, Modal } from "_app/ui";
+
+import { useUALSoftkeyLogin } from "./hooks";
+
+export const UalSoftkeyLoginModal = () => {
+    const { isOpen, dismiss } = useUALSoftkeyLogin();
+    const [fields, setFields] = useFormFields({
+        password: "",
+    });
+    const onChangeFields = (e: React.ChangeEvent<HTMLInputElement>) =>
+        setFields(e);
+
+    const close = () => {
+        alert(
+            "You can't close the authentication modal without confirming or cancelling the password form."
+        );
+    };
+
+    const onCancel = () => {
+        dismiss("");
+    };
+
+    const doSubmit = (e: React.FormEvent) => {
+        e.preventDefault();
+
+        let password: string;
+        let incorrectPassword =
+            "PVT_K1_4nHj22pv211y1CqgaAJLmkPxS688Ha19kPzT2bxLNXMcN7aCb";
+        if (fields.password.length < 12) {
+            password = incorrectPassword; // trigger incorrect password message
```

Comment:
I'm not following this. Why not simply throw "password must be at least 12 chars"? What does having this default PK here means?

Reply:
There's nothing to catch a throw here for this submit handler. So we get an unhandled runtime error. Feeding it a password we know to be incorrect triggers the regular incorrect-password error handling so I figured that was good enough.

Reply:
`onError` FTW. Thanks!